src/alto-ui/Form/DatePicker/DatePicker.js

```diff
@@ -76,16 +76,18 @@ class DatePicker extends React.Component {
 
   handleBlur(e) {
     const { value } = this.state;
-    const { format } = this.props;
+    const { format, displayFormat } = this.props;
 
     if (!value) return null;
 
     if (this.props.inputProps.onBlur) {
       this.props.inputProps.onBlur(e);
     }
 
-    const date = DateTime.fromFormat(value, format);
-
+    let date = DateTime.fromFormat(value, format);
+    if (!date.isValid) {
+      date = DateTime.fromFormat(value, displayFormat);
+    }
     if (date.isValid) {
```

Comment:
A more immutable approch could be:
```js
const date = [format, diplayFormat]
    .map(f => DateTime.fromFormat(value, f))
    .find(d => d.isValid);
if (date) {
```

or perhaps better:

```js
const date = [format, diplayFormat]
    .reduce((d, f) => d && d.isValid ? d : DateTime.fromFormat(value, f), null);
if (date) {
```

Reply:
fixed